wl/server/ClientThread.java

```diff
@@ -40,38 +56,36 @@ public void run() {
 				throw new ServerUtils.WLProtocolException("Stream continues past its end");
 			}
 			boolean admin = false;
+			long userDatabaseID = -1;
 			if (username.isEmpty()) {
 				out.println("ENDOFSTREAM");
 			} else {
 				final long r = ServerUtils.RANDOM.nextLong();
 				out.println(r);
 				out.println("ENDOFSTREAM");
 
-				java.sql.ResultSet sql = database.createStatement().executeQuery(
+				ResultSet sql = ServerUtils.sqlQuery(
+				    ServerUtils.Databases.kWebsite,
 				    "select id from auth_user where username='" + username + "'");
 				if (!sql.next())
 					throw new ServerUtils.WLProtocolException("User " + username +
 					                                          " is not registered");
-				final long userID = sql.getLong​(1);
+				userDatabaseID = sql.getLong​("id");
 
-				sql = database.createStatement().executeQuery(
-				    "select permissions from wlggz_ggzauth where user_id=" + userID);
+				sql = ServerUtils.sqlQuery(
+				    ServerUtils.Databases.kWebsite,
+				    "select permissions,password from wlggz_ggzauth where user_id=" +
+				        userDatabaseID);
 				if (!sql.next())
-					throw new ServerUtils.WLProtocolException("User " + username +
-					                                          " has no permissions at all");
-				final long permissions = sql.getLong(1);
+					throw new ServerUtils.WLProtocolException(
+					    "User " + username + " did not set an online gaming password");
+				final long permissions = sql.getLong("permissions");
 				if (permissions != 7 && permissions != 127)
 					throw new ServerUtils.WLProtocolException(
 					    "User " + username + " has invalid permissions code " + permissions);
 				admin = (permissions == 127);
-
-				sql = database.createStatement().executeQuery(
-				    "select password from wlggz_ggzauth where user_id=" + userID);
-				if (!sql.next())
-					throw new ServerUtils.WLProtocolException("User " + username +
-					                                          " did not set a password");
 				String passwordHash = "";
-				for (byte b : Base64.getDecoder().decode(sql.getString(1)))
+				for (byte b : Base64.getDecoder().decode(sql.getString("password")))
 					passwordHash = String.format("%s%02x", passwordHash, b);
```

Comment:
This used to work in my local setup, but now I need to use `passwordHash = sql.getString("password")` for a correct result... Any ideas what could have changed?

Reply:
I had changed this via a hotfix in 790371a6d066e2e0ef407123ee14ca5ede83ba1a because it turned out that the password hashes are stored in Base64 encoding in the official database. So you need to feed the values of your test database into the `base64` command to get the values the server can parse.